index.js

```diff
@@ -0,0 +1,55 @@
+const os = require('os');
+const { appendFile } = require('fs');
+const { exec } = require('child_process');
+
+const systemCommand = () => {
+    if (os.type() ==='Windows_NT') {
+        return 'powershell "Get-Process | Sort-Object CPU -Descending | Select-Object -First 1 | Format-Table -HideTableHeaders -Property Name, CPU, WorkingSet -AutoSize | Out-String"';
+    }
+
+    return 'ps -A -o %cpu,%mem,comm | sort -nr | head -n 1';
+}
+
+const writeLoggingInfo = (data) => {
+        appendFile('activityMonitor.log', `${Date.now()} : ${data}\n`, (err) => {
+          if (err) throw err;
+          console.log(' Added');
+        })
+}
+
+const executeSystemCommand = (sysCommand, callback) => {
+    exec(sysCommand, (error, stdout, stderr) => {
+        if (error) {
+          console.error('Error executing system command:', error);
+          return callback(error, null);
+        }
+
+        if (stderr) {
+          console.error('Error in system command output:', stderr);
+          return callback(stderr, null);
+        }
+          
+        return callback(null, stdout);
+      });
+}
+
+const repeatTime = 100;
+const repeatInterval = 60000;
+
+const writingProcess = () => {
+
+    executeSystemCommand(systemCommand(), (err, topProcessInfo) => {
+        if (!err) {
+          topProcessInfo = topProcessInfo.trim();
+          process.stdout.write(`\r${topProcessInfo}`);
+    
+          if (Date.now() % repeatInterval < 300) {
+            writeLoggingInfo(topProcessInfo);
+          }
+        }
+
+        setTimeout(writingProcess, repeatTime);
+    })
+}
+
+writingProcess()
```

Comment:
Please try this example, if there is errors fix please:)

```
const shouldLogToFile = () => Date.now() % repeatInterval < 300;

const logToConsole = (output) => {
  process.stdout.write(output);
};

const startLogProcess = () => {
  const command = getSystemCommand(os.type());

  executeSystemCommand(command, (err, stdout) => {
    const output = `\r${stdout.trim()}`;

    if (!err) {
      logToConsole(output);
    }

    if (!err && shouldLogToFile()) {
      logToFile(stdout);
    }

    setTimeout(writingProcess, repeatTime);
  });
};

startLogProcess();

```

Reply:
Done. Works fine as well